### ISSUE TITLE
Every song is greyed out, no playback available

### ISSUE THREAD

**Issue by JimKatsanoss** [abba23/spotify-adblock#80]
I haven't used spotify in a while and suddenly i just could not play anything, every song is greyed out now. I upgraded to the latest so but this didn't fix the issue unfortunately. If any logs can help you out, let me know how i could take them...
Thank you for creating this

**Comment by dswapnil104** [comment]
Apparently the temporary fix was to reinstall incorporating new branch commits as per #60  and #70 .
But that doesn't seem to be the case any more.

- At least on Debain based distros : currently running on PopOS.

### Also here is something weird  I noticed : 

1. On running standalone command from terminal : 

> `$ LD_PRELOAD=/usr/local/lib/spotify-adblock.so spotify`

 and fiddling with some playlists or anything general brings back all greyed-out items back from dead to life .

- Here is terminal Output : [ _https://pastebin.com/wr2q4UT0_ ](url)

2. Not so much luck with the alias :

> _spotify-adblock.desktop.desktop_ 

Would be happy to provide any info and logs needed.

**Comment by dswapnil104** [comment]
Reinstalling from scratch helps as per #70 and #60.

But here is the catch :
There is a noticeable delay before all songs and content comes back to life and becomes accessible for use.
Here is the terminal [**output**](https://pastebin.com/wr2q4UT0) generated on successful execution of command :

- `$ LD_PRELOAD=/usr/local/lib/spotify-adblock.so spotify`

**Comment by abba23** [comment]
>If any logs can help you out, let me know how i could take them...

You can see the log in the terminal you launched Spotify from. Just copy that.

**Comment by ii-Python** [comment]
+1
Fresh install of Linux, running spotify-adblock with spicetify (removing spicetify does **not** solve the problem).
All songs are also grayed out/unavailable.

Attached request logs (running with default config):
```
benjamin ~/ $ LD_PRELOAD=/usr/local/lib/spotify-adblock.so spotify
libva error: /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so init failed
[*] Config file: /etc/spotify-adblock/config.toml
[+] cef_urlrequest_create:	 https://apresolve.spotify.com/?time=1642903770&type=accesspoint
[*] Config file: /etc/spotify-adblock/config.toml
[+] getaddrinfo:		 apresolve.spotify.com
[+] cef_urlrequest_create:	 https://login5.spotify.com/v3/login
[+] getaddrinfo:		 login5.spotify.com
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/user/qw9txd9rcgh87ofhapdtxz209/rootlist?decorate=revision%2Cattributes%2Clength%2Cowner%2Ccapabilities&from=0&length=120
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/facebook/config
[-] cef_urlrequest_create:	 https://spclient.wg.spotify.com/ad-logic/state/config
[-] cef_urlrequest_create:	 https://spclient.wg.spotify.com/ads/v2/config?payload=eyJ1c2VyX2FnZW50IjoiTW96aWxsYVwvNS4wIChYMTE7IExpbnV4IHg4Nl82NCkgQXBwbGVXZWJLaXRcLzUzNi41IChLSFRNTCwgbGlrZSBHZWNrbykgQ2hyb21lXC8xOS4wLjEwODQuOSBTYWZhcmlcLzUzNi41IFNwb3RpZnlcLzExNzIwMDQzOSAoODsgMDsgNSkiLCJzZXNzaW9uX2lkIjoiYzAzMDQ4YTktNzhhZC00MzBmLTg4MjItZmRhNWFiMDgzOGUyIn0=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/recently-played/v3/user/qw9txd9rcgh87ofhapdtxz209/recently-played?limit=50&filter=default,track,collection-new-episodes
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/device-capabilities/v1/capabilities?device_type=computer&client_id=65b708073fc0480ea92a077233ca87bd&device_model=&device_brand=&client_version=1.1.72.439.gc253025e
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/social-connect/v2/sessions/current?alt=protobuf
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/clientsettings/api/v1/
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/user/qw9txd9rcgh87ofhapdtxz209/rootlist/diff?revision=85%2C29fe21fe717dc90028c0e45def6e0dcdf899fa14&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/offline/v1/devices/3630353932306233383934333436336538363134616365616133353265656439/cache/60533031c972fd46204c520fca2c1da8/resources:delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/collection/v2/delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/collection/v2/delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/collection/v2/delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/collection/v2/delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/collection/v2/delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/collection/v2/delta
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/played-state/v1/items?fromTimestamp=1604873207863
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/net-fortune/v2/fortune
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DZ06evO2FvyO4/diff?revision=27380976%2C00000000f1a87304ee1a4d52da3af8e50c7e009b&handlesContent=
[+] getaddrinfo:		 spclient.wg.spotify.com
[+] getaddrinfo:		 spclient.wg.spotify.com
[+] getaddrinfo:		 dealer.spotify.com
[+] getaddrinfo:		 dealer.spotify.com
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/connect-state/v1/devices/7abc32ecf8cd5c08c8abbaa8fedff5ecbf2c6a53
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/played-state/v1/updates
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/connect-state/v1/devices/7abc32ecf8cd5c08c8abbaa8fedff5ecbf2c6a53
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/social-connect/v2/sessions/current?alt=protobuf
[+] getaddrinfo:		 guc3-spclient.spotify.com
[-] cef_urlrequest_create:	 https://spclient.wg.spotify.com/ads/v3/ads?slots=hpto
[+] getaddrinfo:		 api.spotify.com
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/0qw3fh4CHox2ZHkvtd7jUo/diff?revision=20%2Cb4035e49ba2b0101d89ae34efe37418689d728fa&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DWWwaxRea1LWS/diff?revision=1642654800%2C000004c10000017e75db18fd0000017e73c51353&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1EM4UAAg3PWBE5/diff?revision=26774606%2C000000004f272e152f3b3e6a745133131bd2f914&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1EIYFZqHsJ30Wb/diff?revision=1642831%2C000000003ccc35f9555240f9008e9a0082914f37&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DXbcP8BbYEQaO/diff?revision=1642741200%2C000000340000017e7b01755a0000017af7e1de4d&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/0qw3fh4CHox2ZHkvtd7jUo/diff?revision=20%2Cb4035e49ba2b0101d89ae34efe37418689d728fa&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/3EN8bMF81iTHejG9ubIdZS/diff?revision=520%2Cce14c6ed9e48928fe0f7f5382819cca141b5ee6b&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DWSPMbB1kcXmo/diff?revision=1642903638%2C000000000c635f0a7369db713914d27ead164c62&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DZ06evO3Oe9Dq/diff?revision=27380975%2C000000004d734385d798668ae6aac222dedeb036&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DXbYM3nMM0oPk/diff?revision=1642903638%2C00000000bf446cc2cdd6ebf294878f82ae14ecb8&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DX0E9XMGembJo/diff?revision=1642903638%2C00000000e5bafe88abc7d15bcf3e511ad9e8916e&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1E39XhfvjzQfqk/diff?revision=1642850935%2C000000004980e65808cf00e83c0380eee95e0ecd&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1E35wQsk6pUD5U/diff?revision=1642850935%2C000000004796e1b5348e7008334b2e1ee4447763&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1DWTyiBJ6yEqeu/diff?revision=1642903638%2C00000000d0566b71b94186a8498f27828d308074&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1E39Nb02Vvxmjs/diff?revision=1642850935%2C00000000f285fce10ad6f7738b8a05dd022249c0&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1EQn4jwNIohw50/diff?revision=1642831200%2C00000000cd4cbcf6d30922cedfc36a1174b3818b&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/37i9dQZF1E39WlMfEJONFi/diff?revision=1642850935%2C00000000b3cd3aaa731379936c567f0311df7d4c&handlesContent=
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/playlist/v2/playlist/0vvXsWCC9xrXsKd4FyS8kM/diff?revision=2421%2Cf3f0281349bdf351d5f99352dff09086d70ec565&handlesContent=
[+] getaddrinfo:		 misc.scdn.co
[+] getaddrinfo:		 dailymix-images.scdn.co
[+] getaddrinfo:		 api-partner.spotify.com
[+] getaddrinfo:		 seed-mix-image.spotifycdn.com
[+] getaddrinfo:		 newjams-images.scdn.co
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/connect-state/v1/devices/7abc32ecf8cd5c08c8abbaa8fedff5ecbf2c6a53
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/connect-state/v1/devices/7abc32ecf8cd5c08c8abbaa8fedff5ecbf2c6a53
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/connect-state/v1/devices/7abc32ecf8cd5c08c8abbaa8fedff5ecbf2c6a53
[+] cef_urlrequest_create:	 https://spclient.wg.spotify.com/user-customization-service/v1/customize
[-] cef_urlrequest_create:	 https://spclient.wg.spotify.com/gabo-receiver-service/v3/events/
```

**Comment by abba23** [comment]
@ii-Python: 

>libva error: /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so init failed

Sounds like a problem unrelated to spotify-adblock. Have you checked if the problem disappears when launching Spotify normally? Maybe have a look at [this post on askubuntu](https://askubuntu.com/questions/1384489/error-while-running-spotify-libva-error-init-failed).

**Comment by crashship** [comment]
Make sure you have your allowlist in config copied/modified to reflect [this](https://github.com/abba23/spotify-adblock/commit/1388c2d07630e598fc1032a64a414448f7193d37) commit

**Comment by ii-Python** [comment]
After searching for what felt like hours, I can confirm it wasn't a spotify-adblock issue. My bad.

As you suggested, I looked into the libva error and that led me to an infinite driver rabbit hole. In the end, didn't solve the problem. I eventually settled with just searching "linux spotify all songs grayed out" and eventually found [this answer](https://community.spotify.com/t5/Desktop-Linux/Cannot-play-any-file-everything-is-greyed-out/m-p/5327947/highlight/true#M20924) which solved the issue. I'm unsure why `NetworkManager` has anything to do with spotify allowing me to play music, but stopping and disabling it completely solved the issue.

As for spotify-adblock, amazing piece of software you have here.
Cheers :)
